packages/webpack/mixins/fast-build/mixin.core.js

```diff
@@ -0,0 +1,45 @@
+const { Mixin } = require('hops-bootstrap');
+
+function include(rules, path) {
+  const shouldInclude =
+    // allow everything outside of node_modules
+    !path.includes('node_modules') ||
+    // allow all hops packages
+    /node_modules[/\\]hops/.test(path) ||
+    // check if any other rule matches
+    rules.some((rule) => rule.test(path));
+
+  return shouldInclude;
+}
+
+class HopsFastBuildMixin extends Mixin {
+  configureBuild(webpackConfig, { jsLoaderConfig }, target) {
+    const enabled =
+      this.options.fastBuild || (this.options.fastDev && target === 'develop');
```

Comment:
Why do we set `enabled` when we're in `fastDev` mode?

Reply:
Because dev mode would also benefit from less files to transpile. But `target === 'develop'` is actually wrong, we should also do this for `target === 'node'`.

Reply:
I'm still not sure, why we need to handle fast-dev here too, because:
- each mixin should be encapsulated and care only about its own config
- fast-dev and fast-build can never happen at the same time, because fast-dev has an exit clause for target=build.

And if there's something that would also benefit fast-dev we should include it in fast-dev, I think?
Also: We're already excluding all of node_modules in fast-dev: https://github.com/xing/hops/blob/master/packages/webpack/mixins/fast-dev/mixin.core.js#L23

Reply:
> each mixin should be encapsulated and care only about its own config
Sure, but if I would name this `babel-transpile-whitelist` it would be related to `--fast-dev`.
So the issue is more about names than the actual concerns I think.

> fast-dev and fast-build can never happen at the same time, because fast-dev has an exit clause for target=build.

Not for the develop target but we have two others (mock-server and node) that would still benefit from these chages.

I mean we could think about decoupling the flags (fast-dev and fast-build) from the optimizations and just pick from a list of optimizations for each mode...